libs/barista-components/table/src/table.ts

```diff
@@ -370,14 +384,25 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
 
   /** @internal Exports the filtered source data from the dataSource. */
   // Note: this is different from the display text, see instead _exportDisplayData().
-  _exportFilteredData(): void {
-    const exportData = this._filteredData;
-    if (this.isEmptyDataSource || typeof exportData[0] != 'object') {
-      return;
+  _exportFilteredData(selectedData?: T[]): void {
+    //nothing to export
+    if (this.isEmptyDataSource) return;
+    //not using DTDataSource, fallback to this._data instead of this._filteredData
+    let exportData: readonly T[];
```

Comment:
```suggestion
    let exportData: T[];
```

Reply:
This creates a type conflict with CdkTable<T>._data: readonly T[]